plugins/authentication/src/native.cpp

```diff
@@ -33,30 +35,112 @@
 #include <cstring>
 #include <iostream>
 #include <sstream>
+#include <string>
 
-int get64RandomBytes( char *buf );
-void setSessionSignatureClientside( char* _sig );
-void _rsSetAuthRequestGetChallenge( const char* _c );
+void setSessionSignatureClientside(char* _sig);
+#ifdef RODS_SERVER
+int get64RandomBytes(char* buf);
+// NOLINTNEXTLINE(bugprone-reserved-identifier,cert-dcl37-c, cert-dcl51-cpp)
+void _rsSetAuthRequestGetChallenge(const char* _buf);
+#endif
 
-using json = nlohmann::json;
-using log_auth = irods::experimental::log::authentication;
-namespace irods_auth = irods::experimental::auth;
+namespace
+{
+    using json = nlohmann::json;
+    using log_auth = irods::experimental::log::authentication;
+    namespace irods_auth = irods::authentication;
+
+    // Implementation based on clientLoginTTL, which is now deprecated.
+    auto record_limited_password(RcComm& _comm, int ttl, const char* _password) -> void
+    {
+        getLimitedPasswordInp_t inp{};
+        inp.ttl = ttl;
+
+        getLimitedPasswordOut_t* out{};
+        if (const int err = rcGetLimitedPassword(&_comm, &inp, &out); 0 != err) {
+            THROW(err, fmt::format("{}: rcGetLimitedPassword failed with error [{}]", __func__, err));
+        }
+
+        // Calculate the limited password, which is a hash of the user's password and the returned stringToHashWith.
+        constexpr auto hash_length = 100;
+        std::array<char, hash_length + 1> hash_buf{};
+
+        std::strncpy(hash_buf.data(), static_cast<char*>(out->stringToHashWith), hash_length);
+        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc, cppcoreguidelines-owning-memory)
+        std::free(out); // Free the output as it is no longer needed.
+        std::strncat(hash_buf.data(), _password, hash_length);
+
+        std::array<unsigned char, hash_length> digest{};
+        obfMakeOneWayHash(
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            HASH_TYPE_DEFAULT, reinterpret_cast<unsigned char*>(hash_buf.data()), hash_length, digest.data());
+
+        // Clear out the buffer since it exists on the stack and contains a password.
+        std::memset(hash_buf.data(), 0, hash_buf.size());
+
+        std::array<char, hash_length> limited_password{};
+        hashToStr(digest.data(), limited_password.data());
+
+        if (const auto err = obfSavePw(0, 0, 0, limited_password.data()); err < 0) {
+            THROW(err, fmt::format("{}: Failed to save limited password to irodsA file. error: [{}]", __func__, err));
+        }
+    } // record_limited_password
+
+    auto get_password_from_client_stdin() -> std::string
+    {
+        termios tty{};
+        std::memset(&tty, 0, sizeof(tty));
```

Comment:
Can remove this now, I think...

Reply:
If it's POD, then yeah.

Reply:
Yep, this is a C struct: https://linux.die.net/man/3/termios

Done